pkg/integration/docker.go

```diff
@@ -0,0 +1,150 @@
+package integration
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dhis2-sre/im-manager/pkg/config"
+)
+
+func NewDockerHubClient(config config.Config) dockerHubClient {
+	return dockerHubClient{config: config}
+}
+
+type dockerHubClient struct {
+	config config.Config
+}
+
+func (d dockerHubClient) GetTags(organization string, repository string) ([]string, error) {
+	url := fmt.Sprintf("https://hub.docker.com/v2/repositories/%s/%s/tags?page_size=10000", organization, repository)
+
+	token, err := d.getToken()
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("JWT %s", token))
+
+	client := http.Client{}
+
+	response, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	body := struct {
```

Comment:
Itmight make the methods more readable if you declare package level (unexported) types for the payloads like `tagsBody` here.

Reply:
The response struct is the same for both tags and images so I created a struct called `dockerHubResponse` which is used by both.